Cargo.toml

```diff
@@ -13,6 +13,7 @@ edition = "2021"
 name = "ruff"
 
 [dependencies]
+annotate-snippets = { version = "0.9.1", features = ["color"] }
```

Comment:
rustc uses this crate: https://github.com/rust-lang/rust/issues/59346#issuecomment-1279325650

Reply:
Another option is https://github.com/zkat/miette. SWC uses this uses this crate https://github.com/swc-project/swc/blob/27225a0eeb78f3c2975ec14ef51127b2b9532a84/crates/swc_error_reporters/Cargo.toml#L16

Reply:
It looks like that uses https://crates.io/crates/ariadne under the hood.